lib/net/ldap/error.rb

```diff
@@ -0,0 +1,31 @@
+class Net::LDAP
+  class Error < StandardError; end
```

Comment:
@jch I support the rename and keeping around the backwards compatible constant, though the semantics of the backwards compatible constant is a bit trickier than inheriting from `Error`. What needs to be backwards compatible is this:

``` ruby
begin
  # something that could fail
rescue Net::LDAP::LdapError => error
  # handle error
end
```

The problem with the above is that it wouldn't actually catch anything that inherits from `Net::LDAP::Error`, only the specific `Net::LDAP::LdapError` subclass. Instead, I think we can just have a constant assignment that would work as expected:

``` ruby
class Net::LDAP
  class Error < StandardError; end
  LdapError = Error
end

begin
rescue Net::LDAP::LdapError
  # should work
end
```